pcs/src/lib.rs

```diff
@@ -435,8 +438,8 @@ pub mod icicle_deps {
     /// Create a new stream and warmup
     pub fn warmup_new_stream() -> anyhow::Result<CudaStream> {
         let stream = CudaStream::create().map_err(|e| anyhow!("{:?}", e))?;
-        let _warmup_bytes = HostOrDeviceSlice::<'_, u8>::cuda_malloc_async(1024, &stream)
-            .map_err(|e| anyhow!("{:?}", e))?;
+        let _warmup_bytes =
+            DeviceVec::<u8>::cuda_malloc_async(1024, &stream).map_err(|e| anyhow!("{:?}", e))?;
```

Comment:
i think we can forgo this function in favor of out-of-box: https://github.com/ingonyama-zk/icicle/blob/621676bd41659bc101f29dc7ac54f29c7cdaa92b/wrappers/rust/icicle-cuda-runtime/src/device.rs#L44